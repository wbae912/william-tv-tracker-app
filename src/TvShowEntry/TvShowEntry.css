.tv-show-entry {
  border: 1px solid rgb(33, 56, 56);
  background-color: rgb(33, 56, 56);
  border-radius: 10px;
  margin-bottom: 3%;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  cursor: pointer;
}

.checked {
  color: orange;
}

.fa-trash, .fa-edit {
  margin-top: 1%;
  margin-bottom: 2%;
  height: 30px;
  cursor: pointer;
}

.fa-trash {
  margin-left: 8%;
  /* MAKE THIS 2-3% ON BIGGER DEVICES */
}

.fa-edit:hover {
  color: blue;
}

.fa-trash:hover {
  color: red;
}

.p-edit, .p-delete {
  margin: 0;
  margin-top: 1%;
}

.p-edit {
  color: blue;
}

.p-delete {
  color: red;
}