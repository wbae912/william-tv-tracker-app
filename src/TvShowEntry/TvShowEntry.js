import React, { Component } from 'react';
import tvContext from '../Context';
import './TvShowEntry.css';

export default class TvShowEntry extends Component {
  static contextType = tvContext;

  constructor(props) {
    super(props)
  
    this.state = {
       touched: false,
       editHovered: false,
       deleteHovered: false
    }
  }
  
  toggleTouched = (e) => {
    this.setState({
      touched: !this.state.touched
    })
  }

  toggleEditHovered = (e) => {
    this.setState({
      editHovered: true
    })
  }

  toggleDeleteHovered = (e) => {
    this.setState({
      deleteHovered: true
    })
  }

  mouseOutEdit = (e) => {
    this.setState({
      editHovered: false
    })
  }

  mouseOutDelete = (e) => {
    this.setState({
      deleteHovered: false
    })
  }
  
  render() {
    let rating = this.props.show.rating;
    if(rating === 0) {
      rating = 
      <>
        <span className="not-rated">Not Rated</span>
      </>
    }
    if(rating === 1) {
      rating =
      <>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star"></span>
        <span className="fa fa-star"></span>
        <span className="fa fa-star"></span>
        <span className="fa fa-star"></span>
      </>
    }
    if(rating === 2) {
      rating =
      <>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star"></span>
        <span className="fa fa-star"></span>
        <span className="fa fa-star"></span>
      </>
    }
    if(rating === 3) {
      rating =
      <>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star"></span>
        <span className="fa fa-star"></span>
      </>
    }
    if(rating === 4) {
      rating =
      <>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star"></span>
      </>
    }
    if(rating === 5) {
      rating =
      <>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
        <span className="fa fa-star checked"></span>
      </>
    }

    return (
      <div className="tv-show-entry" onClick={e => this.toggleTouched(e)}>
        <h2 className="entry-title">{this.props.show.tv_title}</h2>
        <div className="entry-div">
          <p className="entry-p">Season: {this.props.show.season_number}</p>
          <p className="entry-p">Episode: {this.props.show.episode_number}</p>
          <p className="entry-p">Genre: {this.props.show.genre}</p>
          <p className="entry-p">Rating: {rating}</p>
        </div>

        {this.state.touched
          ? (
            <div className="touched-p">
              <p className="entry-p">Description: {this.props.show.description}</p>
              <p className="entry-p">Review: {this.props.show.review}</p>
             </div>
            )
          : null
        }

        <div className="entry-buttons">
          {this.state.editHovered ? <p className="p-edit">Edit TV Show</p> : null}
          {this.state.deleteHovered ? <p className="p-delete">Delete TV Show</p> : null}
          <i className="fa fa-edit fa-lg" id="edit-button" 
            onClick={() => this.props.history.push(`/edit-entry/${this.props.show.id}`)} 
            onMouseOver={this.toggleEditHovered}
            onMouseOut={this.mouseOutEdit}>
          </i>
          <i className="fa fa-trash fa-lg" aria-hidden="true" id="delete-button" 
            onClick={() => this.context.deleteTvShow(this.props.show.id)}
            onMouseOver={this.toggleDeleteHovered}
            onMouseOut={this.mouseOutDelete}>
          </i>
        </div>
      </div>
    )
  }
}