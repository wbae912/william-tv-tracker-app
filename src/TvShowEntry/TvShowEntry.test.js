import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import TvShowEntry from './TvShowEntry';

// describe(`Completed component`, () => {
//   const show = {
//     "id": "1",
//     "tv_title": "Example",
//     "status": "Completed",
//     "season_number": "1",
//     "episode_number": "1",
//     "rating": "3",
//     "genre": "Action",
//     "description": "Lorem ipsum",
//     "review": "Lorem ipsum"
//   };
  
//   it('renders component without crashing', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(
//       <MemoryRouter>
//         <TvShowEntry props={show}/>
//       </MemoryRouter>
//       , div);
//     ReactDOM.unmountComponentAtNode(div);
//   });
//   it('renders UI as expected', () => {
//     const tree = renderer
//       .create(<MemoryRouter><TvShowEntry props={show}/></MemoryRouter>)
//       .toJSON();
//       expect(tree).toMatchSnapshot();
//   })
//   it('renders a .TvShowEntry by default', () => {
//     const wrapper = shallow(<TvShowEntry props={show}/>)
//     expect(toJson(wrapper)).toMatchSnapshot()
//   })
// })