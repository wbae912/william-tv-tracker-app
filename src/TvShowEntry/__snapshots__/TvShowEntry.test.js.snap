// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowEntry component renders a .TvShowEntry by default 1`] = `
<li
  className="tv-show-entry"
  onClick={[Function]}
>
  <h2
    className="entry-title"
    id="show-entry-h2"
  >
    Example
  </h2>
  <div
    className="entry-div"
  >
    <p
      className="entry-p"
    >
      Season: 
      1
    </p>
    <p
      className="entry-p"
    >
      Episode: 
      1
    </p>
    <p
      className="entry-p"
    >
      Genre: 
      Action
    </p>
    <p
      className="entry-p"
    >
      Rating: 
      3
    </p>
  </div>
  <div
    aria-live="polite"
    className="entry-buttons"
  >
    <i
      aria-label="Edit TV show button"
      className="fa fa-edit fa-lg edit-button"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <i
      aria-label="Delete TV show button"
      className="fa fa-trash fa-lg"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
  </div>
</li>
`;
